### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by unquietwiki** [yglukhov/nimx#450]
I started writing something that I thought nimx could help me out with, but the documentation is pretty sparse. I'm having to switch gears to use https://pysimplegui.readthedocs.io/en/latest/ since I'm on a time crunch, but I suspect the amount of content in this library is comparable to that. At least my Nim code should somewhat carry over to Python, but I really wanted to use Nim. I also ran into last-minute snafus like grabbing copies of SDL2.dll manually just to get it to work; someone not aware of that wouldn't even know how to look for it.

**Goal for this issue:** how to get the documentation for this library, to be more like other well-versed GUI libraries in similar languages?

**Comment by nixfreak** [comment]
I agree, the the library is really good. When I have some time I to would like to write documentation for this.

**Comment by nixfreak** [comment]
Going through the code I have seen both ways of setting up a layout. Either by using the DSL or not. The DSL way as far as I understand is for "layouts that are used frequently". Maybe that isn't the case ?

**Comment by yglukhov** [comment]
Yeah, DSL is the recommended default way to setup the layout. There's a lot of non-dsl code in master branch, just because DSL was introduced later.  version2 branch migrates even more to the dsl way. Also there's an editor in the works by @Tormund. All approaches have their pros and cons.

**Comment by nixfreak** [comment]
Yeah I still can't figure out the editor for Rod. 

I'm still digging through all the code for nimx and rod
